test/run

```diff
@@ -8,26 +8,27 @@ set -eu
 test/common/make-bots
 test/common/pixel-tests pull
 
-TEST_SCENARIO="${TEST_SCENARIO:-verify}"
-[ "${TEST_SCENARIO}" = "${TEST_SCENARIO%%devel}" ] || export NODE_ENV=development
-
-case "$TEST_SCENARIO" in
-    verify|devel|firefox|firefox-devel|pybridge)
-        RUN_OPTS="--track-naughties"
-        PREPARE_OPTS=""
-        [ "${TEST_SCENARIO}" = "${TEST_SCENARIO%%devel}" ] || RUN_OPTS="$RUN_OPTS --coverage"
-        [ "${TEST_SCENARIO}" = "${TEST_SCENARIO%%pybridge}" ] || PREPARE_OPTS="$PREPARE_OPTS --python"
-        [ "${TEST_SCENARIO}" = "${TEST_SCENARIO##firefox}" ] || export TEST_BROWSER=firefox
-
-        test/image-prepare --verbose ${PREPARE_OPTS} "${TEST_OS}"
-        test/common/run-tests --jobs ${TEST_JOBS:-1} --test-dir test/verify ${RUN_OPTS}
-        ;;
-    daily)
-        bots/image-customize --fresh -v --run-command 'dnf -y copr enable rpmsoftwaremanagement/dnf-nightly && dnf -y copr enable @storage/udisks-daily && dnf -y --setopt=install_weak_deps=False update >&2' $TEST_OS
-        test/image-prepare --verbose --overlay $TEST_OS
-        test/common/run-tests --jobs ${TEST_JOBS:-1} --test-dir test/verify --track-naughties
-        ;;
-    *)
-        echo "Unknown test scenario: $TEST_SCENARIO"
-        exit 1
+PREPARE_OPTS=""
+RUN_OPTS=""
+
+# every known case needs to set RUN_OPTS to something non-empty, so that we can check if we hit any branch
+case "${TEST_SCENARIO:-}" in
+    "") RUN_OPTS=" " ;;&
+    *devel*) RUN_OPTS="$RUN_OPTS --coverage"; export NODE_ENV=development ;;&
+    *pybridge*) RUN_OPTS="$RUN_OPTS "; PREPARE_OPTS="$PREPARE_OPTS --python" ;;&
+    *firefox*) RUN_OPTS="$RUN_OPTS "; export TEST_BROWSER=firefox ;;&
+
+    *daily*)
+        bots/image-customize --fresh -v --run-command 'dnf -y copr enable rpmsoftwaremanagement/dnf-nightly && dnf -y copr enable @storage/udisks-daily && dnf -y --setopt=install_weak_deps=False update >&2' "$TEST_OS"
+        RUN_OPTS="$RUN_OPTS "
+        PREPARE_OPTS="$PREPARE_OPTS --overlay"
+        ;;&
 esac
+
+if [ -z "$RUN_OPTS" ]; then
```

Comment:
What about `if [ -n "$TEST_SCENARIO" -a -z "$RUN_OPTS"  ]` and removing the `"")` case above? That would be slightly easier to follow imo. I.e., "if a non-default test scenario is requested, but options are still at their default, we didn't recognize it".

Reply:
Or put the whole `case $TEST_SCENARIO` inside `if [ -n "$TEST_SCENARIO" ]`

Reply:
I did the first option, I agree that's a bit cleaner. I checked that this doesn't break #18949 for what that wants to do.

Reply:
Cheers!